### PR TITLE
Create public app when deploying a hostapp

### PR DESCRIPTION
**Pull request by alexgg**
This will be used by the jenkins build jobs when deploying to staging and production.

### CODE REVIEW

File: automation/include/balena-api.inc [hunk 1]
```diff
@@ -2,8 +2,325 @@
...
+# Returns success if the check fails
+__check_fail() {
+	local _json
+	local _msg
+	_json="$1"
+	_msg="$2"
+
+	if [ "${_json}" != "OK" ]; then
+		__pp_json "${_json}"
+		>&2 echo "${_msg}"
```
Comment (klutchell) [inline automation/include/balena-api.inc:41]:
```suggestion
		_dlog "${_msg}"
```
Unless you intended for this to print when DEBUG is not set?

File: automation/include/balena-api.inc [hunk 2]
```diff
@@ -2,8 +2,325 @@
...
+	while read -r -d '' _post_data <<-'EOF'
+{
+	"is_public": true
+}
+EOF
+do
+	# This avoid read returning error from not finding a newline termination
+	:
+done
+_json=$(${CURL} -XPATCH "https://api.${_apiEnv}/${TRANSLATION}/application(${_appID})" -H "Content-Type: application/json" -H "Authorization: Bearer ${_token}" --data "${_post_data}")
```
Comment (klutchell) [inline automation/include/balena-api.inc:151]: NIT whitespace

File: automation/include/balena-api.inc [hunk 3]
```diff
@@ -2,8 +2,325 @@
...
+	while read -r -d '' _post_data <<-'EOF'
+{
+	"is_host": true
+}
+EOF
+do
+	# This avoid read returning error from not finding a newline termination
+	:
+done
+_json=$(${CURL} -XPATCH "https://api.${_apiEnv}/${TRANSLATION}/application(${_appID})" -H "Content-Type: application/json" -H "Authorization: Bearer ${_token}" --data "${_post_data}")
```
Comment (klutchell) [inline automation/include/balena-api.inc:190]:
```suggestion
	_json=$(${CURL} -XPATCH "https://api.${_apiEnv}/${TRANSLATION}/application(${_appID})" -H "Content-Type: application/json" -H "Authorization: Bearer ${_token}" --data "${_post_data}")
```

File: automation/include/balena-api.inc [hunk 4]
```diff
@@ -2,8 +2,325 @@
...
+	while read -r -d '' _post_data <<-EOF
+{
+	"is_stored_at__repository_url": ${_url}
+}
+EOF
+do
+	# This avoid read returning error from not finding a newline termination
+	:
+done
+_json=$(${CURL} -XPATCH "https://api.${_apiEnv}/${TRANSLATION}/application(${_appID})" -H "Content-Type: application/json" -H "Authorization: Bearer ${_token}" --data "${_post_data}")
```
Comment (klutchell) [inline automation/include/balena-api.inc:258]:
```suggestion
	_json=$(${CURL} -XPATCH "https://api.${_apiEnv}/${TRANSLATION}/application(${_appID})" -H "Content-Type: application/json" -H "Authorization: Bearer ${_token}" --data "${_post_data}")
```

File: automation/include/balena-api.inc [hunk 5]
```diff
@@ -354,3 +364,323 @@ balena_api_is_dt_private() {
 	fi
 	echo "${_is_private}"
 }
+
+# Gets a release
+# Arguments:
+#
+# $1: Application name
+# $2: Release version string
```
Comment (klutchell) [inline automation/include/balena-api.inc:372]: Can we clarify the syntax of the version string in this comment?

File: automation/include/balena-api.inc [hunk 6]
```diff
@@ -47,3 +364,323 @@ balena_api_is_dt_private() {
...
+# $3: Balena target environment
+#
```
Comment (klutchell) [inline automation/include/balena-api.inc:374]:
```suggestion
# $4: Balena API token
#
```

File: automation/include/balena-api.inc [hunk 7]
```diff
@@ -47,3 +364,323 @@ balena_api_is_dt_private() {
...
+}
+
+# Output a space separated list of images labelled for the given store
+#
+# Inputs:
+#
+# $1: Application name
+# $2: Release version
+# $3: Store type
+# $4: Balena API environment (defaults to balena-cloud.com)
```
Comment (klutchell) [inline automation/include/balena-api.inc:403]:
```suggestion
# $4: Balena API environment (defaults to balena-cloud.com)
# $5: Balena API token
```

File: automation/include/balena-api.inc [hunk 8]
```diff
@@ -47,3 +364,323 @@ balena_api_is_dt_private() {
 	fi
 	echo "${_is_private}"
 }
+
+# Gets a release
+# Arguments:
+#
+# $1: Application name
+# $2: Release version string
+# $3: Balena target environment
```
Comment (klutchell) [inline automation/include/balena-api.inc:373]:
```suggestion
# $3: Balena target environment
# $4: Balena API token
```

File: automation/include/balena-api.inc [hunk 9]
```diff
@@ -47,3 +364,323 @@ balena_api_is_dt_private() {
...
+}
+
+# Output a space separated list of images labelled with the given class
+#
+# Inputs:
+#
+# $1: Application name
+# $2: Release version
+# $3: Class type
+# $4: Balena API environment (defaults to balena-cloud.com)
```
Comment (klutchell) [inline automation/include/balena-api.inc:429]:
```suggestion
# $4: Balena API environment (defaults to balena-cloud.com)
# $5: Balena API token
```

File: automation/include/balena-api.inc [hunk 10]
```diff
@@ -47,3 +364,323 @@ balena_api_is_dt_private() {
...
+	echo "${_images}"
+}
+
+# Output a space separated list of labels for a given release version
+#
+# Inputs:
+#
+# $1: Application name
+# $2: Release version
+# $3: Balena API environment (defaults to balena-cloud.com)
```
Comment (klutchell) [inline automation/include/balena-api.inc:492]:
```suggestion
# $3: Balena API environment (defaults to balena-cloud.com)
# $4: Balena API token
```

File: automation/include/balena-api.inc [hunk 11]
```diff
@@ -47,3 +364,323 @@ balena_api_is_dt_private() {
...
+	[ "${_release}" = "null " ] && return
+	_labels=$(echo "${_release}" | jq -r '.d[0].composition.services[].labels' | tr "\n" " ")
+	echo "${_labels}"
+}
+
+# Print application ID from application name
+# Arguments:
+#
+# $1: Application name
+# $2: Balena target environment
```
Comment (klutchell) [inline automation/include/balena-api.inc:514]:
```suggestion
# $2: Balena target environment
# $3: Balena API token
```

File: automation/include/balena-api.inc (hunk 5 above)
Comment (alexgg) [reply automation/include/balena-api.inc:372]: It's going to depend on the app, but I guess I could say that it's usually a semver.

**Comment by alexgg** [comment]
@balena-ci I self-certify!

**Pull request merged**